### PR TITLE
fix: Reduces synchronization.

### PR DESCRIPTION
**Pull request by gpolitis**
Removes some unnecessary synchronized blocks.

### CODE REVIEW

File: src/main/java/org/jitsi/videobridge/Endpoint.java [hunk 1]
```diff
@@ -580,10 +577,10 @@ private void onPinnedEndpointChangedEvent(
         // Find the new pinned endpoint.
         String newPinnedEndpointID = (String) jsonObject.get("pinnedEndpoint");
 
-        List<String> newPinnedIDList = Collections.EMPTY_LIST;
+        Set<String> newPinnedIDList = Collections.EMPTY_SET;
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/Endpoint.java]: maybe we want to change the variable to something like `newPinnedIDs` (or just `newPinnedEndpoints` like used in `pinnedEndpointsChanged` below)?  also...doesn't look like we get to take advantage of multiple endpoints being pinned here, we only parse a single id?

File: src/main/java/org/jitsi/videobridge/Endpoint.java [hunk 2]
```diff
@@ -674,59 +652,67 @@ private void pinnedEndpointsChanged(List<String> pinnedEndpoints)
...
+        if (newSelectedEndpointID != null && !"".equals(newSelectedEndpointID))
         {
-            logger.debug(Logger.Category.STATISTICS,
-                         "selected_message," + getLoggingId()
-                         + " selected=" + newSelectedEndpointIDs);
+            newSelectedIDList = Collections.singleton(newSelectedEndpointID);
         }
 
-        Set<Endpoint> newSelectedEndpoints = new HashSet<>();
+        selectedEndpointsChanged(newSelectedIDList);
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/Endpoint.java]: same question here...we only parse a single one and override any existing, so is the list useful?

File: src/main/java/org/jitsi/videobridge/Endpoint.java (hunk 1 above)
Comment (gpolitis) [inline src/main/java/org/jitsi/videobridge/Endpoint.java]: I should've changed the variable names, thanks @bbaldino. It is true that we only parse a single ID; meet is sending a single ID at a time, but the JVB code is ready to handle the general case where you have a single data channel message with multiple pinned/selected endpoints.

**Comment by gpolitis** [comment]
thanks @bgrozev 👍

**Pull request merged**